### DISCUSSION TITLE
SDR Receiver Brief Survey and Side-by-Side Comparisions

### DISCUSSION THREAD

**Discussion by tkreyche** [jvde-github/AIS-catcher#333] [Show and tell]
### Brief Intro
Besides my AIS Stations, I'm on HF FT8 with my FlexRadio FLEX-8400M as amateur radio operator KF7NOP...currently have the DXCC 150 award, working on 200. There are a many local ADS-B receiving stations, so there is very good coverage. I'm retired from flying sailplanes and spending time sailing, and am now more interested in AIS. Aircraft can be received widely due to their altitude. The geography of Puget Sound makes it challenging to have good AIS coverage, and it's important because there is a significant amount of ship traffic - ferries, cruise ships, tugs, military and commercial vessels.

A comprehensive analysis of some SDR hardware from several years ago is available in the [SDR Makerspace](https://gitlab.com/librespacefoundation/sdrmakerspace/sdreval).
There is also a comparison at [GNU Radio.](https://wiki.gnuradio.org/index.php/Hardware)

### Observations
Here are some basic observations from running different SDR receivers, specifically for AIS receiving with AIS-catcher. These observations use live data, not recorded or simulated. I don't have the test equipment necessary to comprehensively analyze SDR receiver performance.

**RF Splitter**
For comparing two receivers at the same time, I connect the receivers to the antenna using a splitter. It uses the MiniRF SPA-0501-25H splitter. I can't remember where I bought the board...probably on ebay (yes it should be in an enclosure.) AIS-catcher is running on RPi 4Bs. There is about 10 feet of low-loss coax between the antenna and splitter. This equipment is in a loft and swapping anything requires running up and down a ladder. I compare results using the AIS-catcher message count and signal level plots.

<img src="https://github.com/user-attachments/assets/7aaf8ea9-9c91-444f-b6c7-6c21d8a4bea5" width="200">

**RTL-SDR.COM V3**
Probably the standard dongle by which others are measured. Provides decent results, especially for the price. They still seem to be available, supposedly the V4 came out because the V3 receiver IC was going out of production. Jasper has good tuning suggestions on the main page.

**RTL-SDR.COM V4**
I was disappointed to discover that the V4 unit I have gives worse results than the V3 dongle...it's supposed to be better, according to the manufacturer. Other people seem to be using it successfully, so maybe there is a problem with mine. Note that it uses different drivers from V3 and some additional steps may be necessary to install them. [Here's a discussion of the development of the V4.](https://www.rtl-sdr.com/rtl-sdr-blog-v4-dongle-initial-release/)

**Airspy HF+**
Provides ~10% to15% higher AIS message count than my RTL-SDR V3 dongle. Changes to Threshold high/low and Preamp on/off and AFC_WIDE on/off don't seem to make any difference. May not be worth buying as an upgrade over the RTL-SDR V3, unless you'd like a small performance bump or want to experiment with HF, where it works well. I upgraded from several firmware revisions to the current and it didn't seems to make any difference.

**ShipXplorer AIS Dongle**
I have one but it's broken so I can't retest. Other AIS-catcher users have reported good results. Poor user reviews on Amazon should be mostly disregarded since the reviewers typically don't understand how to use the device, and the information provided by the vendor is confusing.

**Uputronics AIS SAW Filtered Preamp**
I have two of these and don't see any improvement in reception when used with a RTL-SDR V3. One of them seems to be defective since it makes reception worse.

Here are the results of a test, where I put one of of these preamps in front of the passive bandpass filter listed below, which feeds into an Airspy HF+. There is a difference is signal level, but no apparent difference in number of messages received. Images were recorded at the exact same time.

<img src="https://github.com/user-attachments/assets/beb2c38c-e2f7-4d9d-a433-5b781e2a1d52" width="300">
<img src="https://github.com/user-attachments/assets/f9afd521-b609-4e08-968a-a2fccdb4f144" width="300">

My station is about 1 km from these 200 m tall towers, which transmit across the entire radio spectrum.

![radio_towers_400](https://github.com/user-attachments/assets/e86930d7-8124-4b92-a4e5-0a16bfda8413)

**RTL-SDR Wideband LNA (and other similar wideband LNAs)**
Probably won't help unless your station is in an RF-quiet area and may just overload the receiver. Your reception will almost certainly be worse if you are near a strong AIS base station or FM/ VHF TV stations. I have this unit and it degrades reception at my location.

**Antennas**
I'm using a homebuilt quarter wave ground plane antenna for my base station. It works very well and cost almost nothing to build. It's a good choice if you like DIY. I use a Nano VNA for tuning. Here's a good resource:
https://m0ukd.com/calculators/quarter-wave-ground-plane-antenna-calculator/#google_vignette

If your antenna has a long cable run, consider mounting the dongle on the antenna mast and run a long active USB cable to your computer, instead of a coax.

**Directional vs. Omni-Directional Antennas**
Your antenna is the most important piece of equipment that affects the number of ships you will receive. Unless you live in a location with ship traffic on multiple sides, a directional antenna will be a better choice. A simple yagi antenna with one reflector and maybe one director will be an improvement over a an omni-directional antenna such as a 1/4 wave verticals, 5/8 wave verticals and 1/4 wave ground plane. There is a great deal of information on amateur-radio web sites about building VHF antennas and it can be confusing - there's a lot of mythology around antennas! They are great DIY projects and require some building skills and test equipment. Commercial marine antennas are expensive.

Typical design for a low-gain, wide-beam directional yagi antenna
<img src="https://github.com/user-attachments/assets/501634a5-4531-49bb-9f26-37dd1461b0ef" width="200">

**Computer Systems**
Raspberry Pi 4 Model B computers with 2 GB of memory work well with AIS-catcher. Running with other programs (such as SignalK Server, OpenCPN or a large PostgreSQL database) on the same system may require more memory and/or a faster computer such as a Pi 5 with NVMe drive.

**Passive AIS Bandpass Filter**
[This filter from GPIO Labs ](https://gpio.com/products/ais-bandpass-filter-160-162-mhz) should help filter out nearby strong VHF FM/TV signals. There is some insertion loss. The vendor has a photo that shows it wired inline with one of their LNAs. They have both wideband LNAs and a filtered AIS LNA. 

I'm running one of these filters in front of my Airspy HF+, without a LNA preamp. So far it appears to filter out intermittent drastic changes in messages per minute. The source of the changes is almost certainly interference from nearby high power transmitters. After watching this for a week or so it's not 100% clear how much the filter is helping...I need to figure out a better way to test it. It may be that the Airspy HF+ has a better front end than the inexpensive dongles.

The images below are from Marine Traffic charts, taken at completely different times, so it's not a direct comparision.

**_Before_**
 <img src="https://github.com/user-attachments/assets/ea186a59-98f4-41ed-a733-f851157939ed" width="400">

**_After_**
 <img src="https://github.com/user-attachments/assets/f3551053-d8ee-478a-8453-9890c4f57bc4" width="400">

**Comment by JRG1956** [comment]
Thanks for the summary.

I've had a similar experience with the Uputronics AIS filtered preamp. In my case the gain would randomly switch on and off as you can see below with the SDR set to max gain. (Station 1217 - the jump in messages per day resulted from moving the antenna to a location with a view of the sea.)
<img width="1272" alt="Screenshot 2024-11-05 at 13 35 49" src="https://github.com/user-attachments/assets/2cce98e8-c7df-42ab-9999-3c4c208acae3">
I assumed that the gain changes were due to a faulty SDR and replaced my Nooelec V5 with an RTL-SDR Blog v4 only to discover that nothing changed. I then swapped out the Uputronics for a Nooelec Wideband LNA - the problem went away and performance was noticeably better. It turns out I'm in a quiet area for unwanted signals near 162MHz and didn't need a filter.

As you observed, I may not need the Wideband LNA - I'm currently running with the RTL-SDR gain fixed around 28 (still experimenting, dropped it to 25.6 this morning).
<img width="1269" alt="Screenshot 2024-11-28 at 08 11 29" src="https://github.com/user-attachments/assets/307830a7-64f5-425b-ac72-62f8167a5999">
The RTL-SDR Blog v4 has been working well for me. Given the problems with the Uputronics, I can't comment on my experience of the relative performance between it and  the Nooelec v5. My AIS receiver is mounted remotely, close to the antenna, and shares a POE Raspberry Pi5 with my ADSB receiver (Airspy mini and Uputronics preamp/filter). I can't easily swap components to test performance, and the Pi5 is at its limit with two receivers, so a splitter to add the Nooelec isn't an option.

**Reply by tkreyche** [reply]
Thanks for the comments and charts! The idea for this thread is to help people make decisions about buying hardware for use with AIS-catcher.

**Comment by markdj57** [comment]
I've been running with the RTL-SDR v3 for some time now, but only in the last 2 weeks have I actually got the aerial mounted outside so I am interested in increasing range whether using a better aerial or SDR or fine tuning the gain etc. I've ordered the v4 now, so I'll do some comparisons and hopefully be able to post here.

Here's the station:
https://aiscatcher.org/stations/details/1852

Distance/reception charts from aiscatcher and marinetraffic with v3 RTL-SDR and Shakespeare AIS antenna 5215-AIS (squatty body whip).

![kuap8kxwf](https://github.com/user-attachments/assets/d0c57015-889a-4606-b241-22a2ba261600)
![coverage chart](https://github.com/user-attachments/assets/7f0a5905-b8ed-4056-b4d9-fed8fe51c036)

**Comment by markdj57** [comment]
I have just added -a 192K to the start of the bat file for starting ais catcher and it seems to have increased the message count substantially. 
![5pm message count increase chart-png](https://github.com/user-attachments/assets/15b19362-2ec1-4ea5-8c92-d0d217f770d1)
I changed the setting at about 17:00 as can clearly be seen in the graph! I haven't changed the hardware yet. If anyone can shed some light on how to optimise this it would be very helpful. 

https://docs.aiscatcher.org/advanced/samplerate/

**Reply by jvde-github** [reply]
Interesting. I would start with `-a 192K -gr rtlagc on tuner auto` and then experiment a bit with tuner settings to see if you can improve from this base case (192K for bandwidth would keep, reasonable setting). I don't think the  V4 will give a uplift vs the V3. The V3 is surprisingly good for AIS but interested in your experience.

Given that the 192K bandwidth works so well for you, perhaps a filter might help as well. I had some good experience with this [one](https://gpio.com/products/ais-bandpass-filter-160-162-mhz-in-enclosure) and the shipxplorer dongle. But does not improve for everyone. The problem with the filter is that it is quite expensive, specially with freight and taxes. 

Are you using AIS-catcher for decoding. Might be worthwhile to have a look at the signal strength charts.

**Reply by jvde-github** [reply]
See that the GPIO is 42$ which is a bit over the top, can buy a dongle with filter for that money.
